x/logic/predicate/bank_test.go

```diff
@@ -144,7 +145,8 @@ func TestBank(t *testing.T) {
 				balances:   []bank.Balance{},
 				query:      `bank_balances('foo', X).`,
 				wantResult: []types.TermResults{{"X": "[uknow-100]"}},
-				wantError:  fmt.Errorf("bank_balances/2: decoding bech32 failed: invalid bech32 string length 3"),
+				wantError: fmt.Errorf("error(resource_error(resource_module(bank)),[%s],unknown)",
+					strings.Join(strings.Split("decoding bech32 failed: invalid bech32 string length 3", ""), ",")),
```

Comment:
No, it's not exactly the same. In this case, we transform the `string` into a string where each character is separated by a comma. This step is essential because the Prolog interpreter treats error messages as lists of characters. While I could set the string as static, it would make editing and maintaining error messages more cumbersome, imho.